SimplenoteUITests/NotesList.swift

```diff
@@ -137,27 +137,33 @@ class NoteListAssert {
         XCTAssertEqual(actualNotesNumber, expectedNotesNumber, numberOfNotesInAllNotesNotExpected)
     }
 
-    class func allNotesScreenShown() {
-        print(assertNavBarIdentifier + UID.NavBar.allNotes)
-        XCTAssertTrue(app.navigationBars[UID.NavBar.allNotes].waitForExistence(timeout: maxLoadTimeout))
-        XCTAssertEqual(NoteList.getNavBarIdentifier(), UID.NavBar.allNotes, UID.NavBar.allNotes + navBarNotFound)
+    class func noteListShown(forSelection selection: String ) {
+        print(assertNavBarIdentifier + selection)
+        XCTAssertTrue(app.navigationBars[selection].waitForExistence(timeout: maxLoadTimeout))
+
+        if let navBarID = NoteList.getNavBarIdentifier() {
+            XCTAssertEqual(navBarID, selection, selection + navBarNotFound)
+        } else {
+            XCTAssertTrue(false, foundNoNavBar)
+        }
```

Comment:
`XCTAssertTrue(false)` is a neat trick to force a test failure. There's a dedicated API that you can use for that, though:

```suggestion
        if let navBarID = NoteList.getNavBarIdentifier() {
            XCTAssertEqual(navBarID, selection, selection + navBarNotFound)
        } else {
            XCTAssertFail(foundNoNavBar)
        }
```

Another cool XCTest API is `XCTUnwrap`; you can use it to unwrap optional values and fail the test if the value is `.none`.

I haven't tried the code in the IDE, but it should work out to something like:

```suggestion
		let navBarId = try XCTUnwrap(NoteList.getNavBarIdentifier(), foundNoNavBar)
        XCTAssertEqual(navBarID, selection, selection + navBarNotFound)
```

Reply:
Thank you @mokagio, it definitely makes sense for me to explore a bit more for alternatives before writing certain code.

I noticed that using `try XCTUnwrap...` results in a chain of changes: marking all subsequent methods with `throws`, and also prefixing all calls with `try`, including the tests code. For now, I went for your first suggestion of using `XCTFail()`.

Reply:
Good point. I didn't take into account the fact that this is not a test method, that is, the place where I'd usually call `XCTUnwrap`.

Reply:
I generally have some amount of questions/places in my code I don't like that much, but I'd like to firstly discuss how to proceed with it. I thought of maybe having a short call at some point?

Reply:
Sure, happy to give my two cents.

What do you think of writing down all of them (or the subset that can be easily written done) in a P2 (it's a shame to not do it in the open, but P2 has got a better threading support than GitHub) so I can start thinking about them first?